### PR TITLE
Update Ssh2 sftp client type definition to v1.1.0

### PR DESCRIPTION
**Pull request by ascariandrea**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [ssh2-sftp-client](https://github.com/jyu213/ssh2-sftp-client/blob/master/src/index.js#L63)
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by igrayson** [comment]
:-1: This breaks existing consumers.

My code today:
```typescript
import * as SftpClient from 'ssh2-sftp-client';

export default class Foobar {
  private client:SftpClient.Client;
 // ..
}
```

```
$ tsc

29   private client:SftpClient.Client;
                               ~~~~~~

src/lib/clients/foobar.ts(29,29): error TS2694: Namespace '"/Users/ian/workspace/foobar/node_modules/@types/ssh2-sftp-client/index"' has no exported member 'Client'.
```

**Comment by ascariandrea** [comment]
@igrayson the breaks of current usage of this type doesn't mean it's incorrect.

The before type definition didn't reflect the `ssh2-sftp-client` module properly as there's no namespaced `Client` in the package. Thus, also forced by the linter, I add to change the exported type a little bit.

**Comment by ascariandrea** [comment]
@igrayson What's your feedback?

**Comment by igrayson** [comment]
> as there's no namespaced Client in the package

Could you rephrase this? The package is a javascript library, and has no namespaces at all.

**Comment by ascariandrea** [comment]
Yep, that's why we don't need to refer the `client` as a `SftpClient.Client`.
If you look at the example I've changed, the new client instance is now: `const client: new Client()`, without namespacing.

**Comment by igrayson** [comment]
Got it. That sounds fine (and desirable). 

But I don't think it justifies breaking existing clients without bumping a major version (which is a semantic that `@types` kind of struggles with).

How about exporting the type under both?

**Comment by ascariandrea** [comment]
It could be a nice workaround, but I'm not sure how these changes are handled usually.

Also, are you thinking about removing the deprecated way once `ssh2-ftp-client` release the v2?

File: types/ssh2-sftp-client/index.d.ts [hunk 1]
```diff
@@ -1,47 +1,36 @@
...
+    get(remoteFilePath: string, useCompression?: boolean, encoding?: string): Promise<NodeJS.ReadableStream>;
+    put(input: string | Buffer | NodeJS.ReadableStream, remoteFilePath: string, useCompression?: boolean, encoding?: string): Promise<void>;
+    mkdir(remoteFilePath: string, recursive?: boolean): Promise<void>;
+    delete(remoteFilePath: string): Promise<void>;
+    rename(remoteSourcePath: string, remoteDestPath: string): Promise<void>;
+    end(): Promise<void>;
+  }
 }
 
+export default sftp.Client;
```
Comment (RyanCavanaugh) [inline types/ssh2-sftp-client/index.d.ts]: This module doesn't have a default export.

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

File: types/ssh2-sftp-client/index.d.ts (hunk 1 above)
Comment (ascariandrea) [reply types/ssh2-sftp-client/index.d.ts]: Should be good now
